rmw_fastrtps_cpp/include/rmw_fastrtps_cpp/TypeSupport_impl.h

```diff
@@ -513,7 +574,18 @@ void deserialize_array<std::string>(
 {
     (void)call_new;
     if (member->array_size_ && !member->is_upper_bound_) {
-        deser.deserializeArray(((rosidl_generator_c__String*)field)->data, member->array_size_);
+        auto deser_field = (rosidl_generator_c__String*)field;
+        std::vector<std::string> stringseq(member->array_size_);
```

Comment:
updated using a single temporary std::string for storage https://github.com/eProsima/ROS-RMW-Fast-RTPS-cpp/pull/58/commits/b04928bc0fe9b9b4b789cf7bc54fd9b6eef01b7f